lib/child_process.js

```diff
@@ -645,6 +628,23 @@ function spawn(file, args, options) {
     });
   }
 
+  if (options.signal) {
+    const signal = options.signal;
+    if (signal.aborted) {
+      process.nextTick(() => {
+        onAbortListener();
+      });
```

Comment:
nit: we can avoid creating a closure here:
```suggestion
      process.nextTick(onAbortListener);
```